endpoints/openrtb2/auction.go

```diff
@@ -192,12 +193,14 @@ func (deps *endpointDeps) Auction(w http.ResponseWriter, r *http.Request, _ http
 
 	req, impExtInfoMap, storedAuctionResponses, storedBidResponses, bidderImpReplaceImp, account, errL := deps.parseRequest(r, &labels, hookExecutor)
 	if errortypes.ContainsFatalError(errL) && writeError(errL, w, &labels) {
+		seatNonBid.MergeNonBids(getNonBidsFromStageOutcomes(hookExecutor.GetOutcomes()))
```

Comment:
Can we move this to defer on condition if `ao.response == nil`? That means we there is an error or the auction response is empty read all seatnonbids from the hook stage and set it to auctionObject.

Reply:
yes, moved this call to defer